src/url.rs

```diff
@@ -314,6 +314,45 @@ impl PyMultiHostUrl {
     fn __getnewargs__(&self) -> (String,) {
         (self.__str__(),)
     }
+
+    #[classmethod]
+    #[pyo3(signature=(*, scheme, host, user=None, password=None, port=None, path=None, query=None, fragment=None))]
+    #[allow(clippy::too_many_arguments)]
+    pub fn build(
+        _cls: &PyType,
+        scheme: String,
+        host: String,
+        user: Option<String>,
+        password: Option<String>,
+        port: Option<String>,
+        path: Option<String>,
+        query: Option<String>,
+        fragment: Option<String>,
+    ) -> String {
+        let user_password = match (user, password) {
+            (Some(user), None) => format!("{user}@"),
+            (None, Some(password)) => format!(":{password}@"),
+            (Some(user), Some(password)) => format!("{user}:{password}@"),
+            (None, None) => String::new(),
+        };
+        let _port = match port {
```

Comment:
It would be faster (not really important) and more much readable/correct I think to do something like:

```py
        let mut url = format!("{scheme}://{user_password}{host}");
        if let Some(port) = port {
            url.push(':');
            url.push_str(port);
        }
        if let Some(path) = path {
            url.push_str(path);
        }
        // ...
        url
```